### PR TITLE
Fix C99 (and other) maths calls.

### PR DESCRIPTION
**Pull request by stuartarchibald**
As title.

### CODE REVIEW

**Comment by stuartarchibald** [comment]
Maybe fixes https://github.com/numba/numba/issues/5562

File: numba/cpython/mathimpl.py [hunk 1]
```diff
@@ -355,24 +355,17 @@ def hypot_float_impl(context, builder, sig, args):
...
     fname = {
-        types.float32: "_hypotf" if sys.platform == 'win32' else "hypotf",
-        types.float64: "_hypot" if sys.platform == 'win32' else "hypot",
+        types.float32: "hypotf",
+        types.float64: "hypot",
     }[xty]
     plat_hypot = types.ExternalFunction(fname, sig)
 
     if sys.platform == 'win32' and config.MACHINE_BITS == 32:
         inf = xty(float('inf'))
```
Comment (eric-wieser) [inline numba/cpython/mathimpl.py:365]: This `inf` is no longer used

**Comment by stuartarchibald** [comment]
Closing this, there needs to be a more wide spread removal of win32 related branches at some point.

**Pull request closed without merging**